### ISSUE TITLE
Build System Broken in Master

### ISSUE THREAD

**Issue by blink1073** [jupyterlab/jupyterlab#7264]
## Description
Currently, running the APOD extension instructions fail.
When you run `jupyter labextension install .`, in puts the associated `settings/`, `staging/`, `static/` folders in the *extension* dir instead of the the *app* dir, and the extension is not loaded in the application.

## Reproduce

```bash
cd /tmp
git clone https://github.com/jupyterlab/jupyterlab
pip install -e jupyterlab
cookiecutter https://github.com/jupyterlab/extension-cookiecutter-ts --checkout v1.0
cd myextension
yarn
jupyter labextension install .
jupyter lab
```

```bash
(base) silvester: /tmp/myextension 
$ ls
extensions   lib	   package.json  schemas   src	    static  themes	   tsconfig.tsbuildinfo
imports.css  node_modules  README.md	 settings  staging  style   tsconfig.json  yarn.lock
```

## Expected behavior
Extension installs to app dir and prints to the console at startup.

## Context
I think this is related to #7079 

- Operating System and version:
- Browser and version: Firefox 68
- JupyterLab version: 99f2c3ba339443ab6fc3b5ea8d4e567b176b5388

<details><summary>Troubleshoot Output</summary>
<pre>
$PATH:
	/home/silvester/miniconda/bin
	/home/silvester/miniconda/bin
	/home/silvester/miniconda/condabin
	/home/silvester/bin
	/usr/local/sbin
	/usr/local/bin
	/usr/sbin
	/usr/bin
	/sbin
	/bin
	/usr/games
	/usr/local/games

sys.path:
	/home/silvester/miniconda/bin
	/home/silvester/miniconda/lib/python37.zip
	/home/silvester/miniconda/lib/python3.7
	/home/silvester/miniconda/lib/python3.7/lib-dynload
	/home/silvester/miniconda/lib/python3.7/site-packages
	/home/silvester/workspace/jupyter/jupyterlab_server
	/home/silvester/workspace/jupyter/lab

sys.executable:
	/home/silvester/miniconda/bin/python3.7

sys.version:
	3.7.3 (default, Mar 27 2019, 22:11:17) 
	[GCC 7.3.0]

platform.platform():
	Linux-4.15.0-60-generic-x86_64-with-debian-buster-sid

which -a jupyter:
	/home/silvester/miniconda/bin/jupyter
	/home/silvester/miniconda/bin/jupyter

pip list:
	Package                Version   Location                                           
	---------------------- --------- ---------------------------------------------------
	altair                 3.2.0     
	arrow                  0.15.2    
	asn1crypto             0.24.0    
	attrs                  19.1.0    
	backcall               0.1.0     
	binaryornot            0.4.4     
	bleach                 3.1.0     
	certifi                2019.6.16 
	cffi                   1.12.3    
	chardet                3.0.4     
	Click                  7.0       
	conda                  4.7.11    
	conda-package-handling 1.3.11    
	cookiecutter           1.6.0     
	cryptography           2.7       
	cycler                 0.10.0    
	decorator              4.4.0     
	defusedxml             0.5.0     
	docutils               0.15.2    
	entrypoints            0.3       
	future                 0.17.1    
	idna                   2.8       
	ipykernel              5.1.2     
	ipyparallel            6.2.4     
	ipython                7.7.0     
	ipython-genutils       0.2.0     
	ipywidgets             7.5.1     
	jedi                   0.15.1    
	Jinja2                 2.10.1    
	jinja2-time            0.2.0     
	json5                  0.8.5     
	jsonschema             3.0.2     
	jupyter-client         5.3.1     
	jupyter-core           4.4.0     
	jupyterlab             1.2.0a0   /home/silvester/workspace/jupyter/lab              
	jupyterlab-server      1.0.4     /home/silvester/workspace/jupyter/jupyterlab_server
	kiwisolver             1.1.0     
	libarchive-c           2.8       
	loghub                 0.4.0     
	MarkupSafe             1.1.1     
	matlab-kernel          0.16.7    
	matplotlib             3.1.1     
	metakernel             0.24.2    
	mistune                0.8.4     
	nbconvert              5.6.0     
	nbformat               4.4.0     
	notebook               6.0.0     
	numpy                  1.17.2    
	pandas                 0.25.1    
	pandocfilters          1.4.2     
	parso                  0.5.1     
	pexpect                4.7.0     
	pickleshare            0.7.5     
	pip                    19.1.1    
	pkginfo                1.5.0.1   
	poyo                   0.5.0     
	prometheus-client      0.7.1     
	prompt-toolkit         2.0.9     
	ptyprocess             0.6.0     
	pycosat                0.6.3     
	pycparser              2.19      
	Pygments               2.4.2     
	pyOpenSSL              19.0.0    
	pyparsing              2.4.2     
	pyrsistent             0.15.4    
	PySocks                1.7.0     
	python-dateutil        2.8.0     
	pytz                   2019.2    
	pyzmq                  18.0.2    
	readme-renderer        24.0      
	requests               2.22.0    
	requests-toolbelt      0.9.1     
	ruamel-yaml            0.15.46   
	Send2Trash             1.5.0     
	setuptools             41.0.1    
	six                    1.12.0    
	terminado              0.8.2     
	testpath               0.4.2     
	toolz                  0.10.0    
	tornado                6.0.3     
	tqdm                   4.32.1    
	traitlets              4.3.2     
	twine                  1.13.0    
	urllib3                1.24.2    
	vega-datasets          0.7.0     
	wcwidth                0.1.7     
	webencodings           0.5.1     
	wheel                  0.33.4    
	whichcraft             0.6.1     
	widgetsnbextension     3.5.1     
	wurlitzer              1.0.3

conda list:
	# packages in environment at /home/silvester/miniconda:
	#
	# Name                    Version                   Build  Channel
	_libgcc_mutex             0.1                        main  
	altair                    3.2.0                    py37_0    conda-forge
	arrow                     0.15.2                   pypi_0    pypi
	asn1crypto                0.24.0                   py37_0  
	attrs                     19.1.0                     py_0    conda-forge
	backcall                  0.1.0                      py_0    conda-forge
	binaryornot               0.4.4                    pypi_0    pypi
	bleach                    3.1.0                      py_0    conda-forge
	bzip2                     1.0.8                h7b6447c_0  
	ca-certificates           2019.6.16            hecc5488_0    conda-forge
	certifi                   2019.6.16                py37_1    conda-forge
	cffi                      1.12.3           py37h2e261b9_0  
	chardet                   3.0.4                    py37_1  
	click                     7.0                      pypi_0    pypi
	conda                     4.7.11                   py37_0    conda-forge
	conda-package-handling    1.3.11                   py37_0  
	cookiecutter              1.6.0                    pypi_0    pypi
	cryptography              2.7              py37h1ba5d50_0  
	cycler                    0.10.0                     py_1    conda-forge
	dbus                      1.13.6               he372182_0    conda-forge
	decorator                 4.4.0                      py_0    conda-forge
	defusedxml                0.5.0                      py_1    conda-forge
	docutils                  0.15.2                   pypi_0    pypi
	entrypoints               0.3                   py37_1000    conda-forge
	expat                     2.2.5             he1b5a44_1003    conda-forge
	fontconfig                2.13.1            h86ecdb6_1001    conda-forge
	freetype                  2.10.0               he983fc9_1    conda-forge
	future                    0.17.1                   pypi_0    pypi
	gettext                   0.19.8.1          hc5be6a0_1002    conda-forge
	glib                      2.58.3            h6f030ca_1002    conda-forge
	gst-plugins-base          1.14.5               h0935bb2_0    conda-forge
	gstreamer                 1.14.5               h36ae1b5_0    conda-forge
	icu                       64.2                 he1b5a44_0    conda-forge
	idna                      2.8                      py37_0  
	ipykernel                 5.1.2            py37h5ca1d4c_0    conda-forge
	ipyparallel               6.2.4                    pypi_0    pypi
	ipython                   7.7.0            py37h5ca1d4c_0    conda-forge
	ipython_genutils          0.2.0                      py_1    conda-forge
	ipywidgets                7.5.1                      py_0    conda-forge
	jedi                      0.15.1                   py37_0    conda-forge
	jinja2                    2.10.1                     py_0    conda-forge
	jinja2-time               0.2.0                    pypi_0    pypi
	jpeg                      9c                h14c3975_1001    conda-forge
	json5                     0.8.5                    pypi_0    pypi
	jsonschema                3.0.2                    py37_0    conda-forge
	jupyter_client            5.3.1                      py_0    conda-forge
	jupyter_core              4.4.0                      py_0    conda-forge
	jupyterlab                1.2.0a0                   dev_0    <develop>
	jupyterlab-server         1.0.4                     dev_0    <develop>
	kiwisolver                1.1.0            py37hc9558a2_0    conda-forge
	libarchive                3.3.3                h5d8350f_5  
	libblas                   3.8.0               12_openblas    conda-forge
	libcblas                  3.8.0               12_openblas    conda-forge
	libedit                   3.1.20181209         hc058e9b_0  
	libffi                    3.2.1                hd88cf55_4  
	libgcc-ng                 9.1.0                hdf63c60_0  
	libgfortran-ng            7.3.0                hdf63c60_0  
	libiconv                  1.15              h516909a_1005    conda-forge
	liblapack                 3.8.0               12_openblas    conda-forge
	libopenblas               0.3.7                h6e990d7_1    conda-forge
	libpng                    1.6.37               hed695b0_0    conda-forge
	libsodium                 1.0.17               h516909a_0    conda-forge
	libstdcxx-ng              9.1.0                hdf63c60_0  
	libuuid                   2.32.1            h14c3975_1000    conda-forge
	libuv                     1.30.1               h516909a_0    conda-forge
	libxcb                    1.13              h14c3975_1002    conda-forge
	libxml2                   2.9.9                hea5a465_1  
	loghub                    0.4.0                    pypi_0    pypi
	lz4-c                     1.8.1.2              h14c3975_0  
	lzo                       2.10                 h49e0be7_2  
	markupsafe                1.1.1            py37h14c3975_0    conda-forge
	matlab-kernel             0.16.7                   pypi_0    pypi
	matplotlib                3.1.1                    py37_1    conda-forge
	matplotlib-base           3.1.1            py37he7580a8_1    conda-forge
	metakernel                0.24.2                   pypi_0    pypi
	mistune                   0.8.4           py37h14c3975_1000    conda-forge
	nbconvert                 5.6.0                      py_0    conda-forge
	nbformat                  4.4.0                      py_1    conda-forge
	ncurses                   6.1                  he6710b0_1  
	nodejs                    12.8.1               h10a4023_0    conda-forge
	notebook                  6.0.0                    py37_0    conda-forge
	numpy                     1.17.2           py37h95a1406_0    conda-forge
	openssl                   1.1.1c               h516909a_0    conda-forge
	pandas                    0.25.1           py37hb3f55d8_0    conda-forge
	pandoc                    2.7.3                         0    conda-forge
	pandocfilters             1.4.2                      py_1    conda-forge
	parso                     0.5.1                      py_0    conda-forge
	pcre                      8.41              hf484d3e_1003    conda-forge
	pexpect                   4.7.0                    py37_0    conda-forge
	pickleshare               0.7.5                 py37_1000    conda-forge
	pip                       19.1.1                   py37_0  
	pkginfo                   1.5.0.1                  pypi_0    pypi
	poyo                      0.5.0                    pypi_0    pypi
	prometheus_client         0.7.1                      py_0    conda-forge
	prompt_toolkit            2.0.9                      py_0    conda-forge
	pthread-stubs             0.4               h14c3975_1001    conda-forge
	ptyprocess                0.6.0                   py_1001    conda-forge
	pycosat                   0.6.3            py37h14c3975_0  
	pycparser                 2.19                     py37_0  
	pygments                  2.4.2                      py_0    conda-forge
	pyopenssl                 19.0.0                   py37_0  
	pyparsing                 2.4.2                      py_0    conda-forge
	pyqt                      5.9.2            py37hcca6a23_3    conda-forge
	pyrsistent                0.15.4           py37h516909a_0    conda-forge
	pysocks                   1.7.0                    py37_0  
	python                    3.7.3                h0371630_0  
	python-dateutil           2.8.0                      py_0    conda-forge
	python-libarchive-c       2.8                     py37_11  
	pytz                      2019.2                     py_0    conda-forge
	pyzmq                     18.0.2           py37h1768529_2    conda-forge
	qt                        5.9.7                h0c104cb_3    conda-forge
	readline                  7.0                  h7b6447c_5  
	readme-renderer           24.0                     pypi_0    pypi
	requests                  2.22.0                   py37_0  
	requests-toolbelt         0.9.1                    pypi_0    pypi
	ruamel_yaml               0.15.46          py37h14c3975_0  
	send2trash                1.5.0                      py_0    conda-forge
	setuptools                41.0.1                   py37_0  
	sip                       4.19.8          py37hf484d3e_1000    conda-forge
	six                       1.12.0                   py37_0  
	sqlite                    3.29.0               h7b6447c_0  
	terminado                 0.8.2                    py37_0    conda-forge
	testpath                  0.4.2                   py_1001    conda-forge
	tk                        8.6.9             hed695b0_1002    conda-forge
	toolz                     0.10.0                     py_0    conda-forge
	tornado                   6.0.3            py37h516909a_0    conda-forge
	tqdm                      4.32.1                     py_0  
	traitlets                 4.3.2                 py37_1000    conda-forge
	twine                     1.13.0                   pypi_0    pypi
	urllib3                   1.24.2                   py37_0  
	vega_datasets             0.7.0                      py_0    conda-forge
	wcwidth                   0.1.7                      py_1    conda-forge
	webencodings              0.5.1                      py_1    conda-forge
	wheel                     0.33.4                   py37_0  
	whichcraft                0.6.1                    pypi_0    pypi
	widgetsnbextension        3.5.1                    py37_0    conda-forge
	wurlitzer                 1.0.3                    pypi_0    pypi
	xorg-libxau               1.0.9                h14c3975_0    conda-forge
	xorg-libxdmcp             1.1.3                h516909a_0    conda-forge
	xz                        5.2.4                h14c3975_4  
	yaml                      0.1.7                had09818_2  
	zeromq                    4.3.2                he1b5a44_2    conda-forge
	zlib                      1.2.11               h7b6447c_3  
	zstd                      1.3.7                h0b5b093_0
</pre>
</details>

<details><summary>Command Line Output</summary>
<pre>
$ jupyter lab --debug
[D 19:26:04.557 LabApp] Searching ['/tmp/myextension', '/home/silvester/.jupyter', '/home/silvester/miniconda/etc/jupyter', '/usr/local/etc/jupyter', '/etc/jupyter'] for config files
[D 19:26:04.557 LabApp] Looking for jupyter_config in /etc/jupyter
[D 19:26:04.557 LabApp] Looking for jupyter_config in /usr/local/etc/jupyter
[D 19:26:04.558 LabApp] Looking for jupyter_config in /home/silvester/miniconda/etc/jupyter
[D 19:26:04.558 LabApp] Looking for jupyter_config in /home/silvester/.jupyter
[D 19:26:04.558 LabApp] Looking for jupyter_config in /tmp/myextension
[D 19:26:04.558 LabApp] Looking for jupyter_notebook_config in /etc/jupyter
[D 19:26:04.558 LabApp] Looking for jupyter_notebook_config in /usr/local/etc/jupyter
[D 19:26:04.558 LabApp] Looking for jupyter_notebook_config in /home/silvester/miniconda/etc/jupyter
[D 19:26:04.558 LabApp] Looking for jupyter_notebook_config in /home/silvester/.jupyter
[D 19:26:04.558 LabApp] Loaded config file: /home/silvester/.jupyter/jupyter_notebook_config.py
[D 19:26:04.559 LabApp] Looking for jupyter_notebook_config in /tmp/myextension
[D 19:26:04.562 LabApp] Paths used for configuration of jupyter_notebook_config: 
    	/etc/jupyter/jupyter_notebook_config.json
[D 19:26:04.562 LabApp] Paths used for configuration of jupyter_notebook_config: 
    	/usr/local/etc/jupyter/jupyter_notebook_config.json
[D 19:26:04.563 LabApp] Paths used for configuration of jupyter_notebook_config: 
    	/home/silvester/miniconda/etc/jupyter/jupyter_notebook_config.d/ipyparallel-serverextension.json
    	/home/silvester/miniconda/etc/jupyter/jupyter_notebook_config.json
[D 19:26:04.563 LabApp] Paths used for configuration of jupyter_notebook_config: 
    	/home/silvester/.jupyter/jupyter_notebook_config.json
[I 19:26:04.740 LabApp] Loading IPython parallel extension
[W 19:26:04.742 LabApp] JupyterLab server extension not enabled, manually loading...
[I 19:26:06.415 LabApp] JupyterLab extension loaded from /home/silvester/workspace/jupyter/lab/jupyterlab
[I 19:26:06.415 LabApp] JupyterLab application directory is /home/silvester/miniconda/share/jupyter/lab
[D 19:26:08.079 LabApp] Yarn configuration loaded.
[I 19:26:08.081 LabApp] Serving notebooks from local directory: /tmp/myextension

</pre>
</details>

<details><summary>Browser Output</summary>
<pre>
Starting application in workspace: "/lab"
</pre>
</details>

**Comment by blink1073** [comment]
cc @vidartf

**Comment by telamonian** [comment]
I saw this happen about 2 weeks ago as well. For me, nuking the `venv` I was working in and recreating it from scratch fixed the problem. I'll see if I can reproduce this with the current master.

**Comment by telamonian** [comment]
I came up with a fix in #7268. The problem seems to be that if `app_dir=''` is passed to `AppOptions`, it will use that value instead of the default (`get_app_dir()`). So if you don't explicitly configure `app_dir`, the build just assumes that the dir you're in is the Jlab application directory.
